yoti-sdk-impl/src/test/java/com/yoti/api/client/spi/remote/call/aml/RemoteAmlServiceTest.java

```diff
@@ -0,0 +1,178 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.yoti.api.client.AmlException;
+import com.yoti.api.client.aml.AmlProfile;
+import com.yoti.api.client.spi.remote.call.ResourceException;
+import com.yoti.api.client.spi.remote.call.ResourceFetcher;
+import com.yoti.api.client.spi.remote.call.UrlConnector;
+import com.yoti.api.client.spi.remote.call.YotiConstants;
+import com.yoti.api.client.spi.remote.call.factory.PathFactory;
+import com.yoti.api.client.spi.remote.call.factory.SignatureFactory;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.Map;
+
+@RunWith(MockitoJUnitRunner.class)
+public class RemoteAmlServiceTest {
+
+    private static final String SOME_APP_ID = "someAppId";
+    private static final String GENERATED_PATH = "generatedPath";
+    private static final String SERIALIZED_BODY = "serializedBody";
+    private static final String SOME_SIGNATURE = "someSignature";
+
+    @InjectMocks RemoteAmlService testObj;
+
+    @Mock PathFactory pathFactoryMock;
+    @Mock ObjectMapper objectMapperMock;
+    @Mock ResourceFetcher resourceFetcherMock;
+    @Mock SignatureFactory signatureFactoryMock;
+
+    @Mock AmlProfile amlProfileMock;
+    KeyPair keyPair;
+    @Captor ArgumentCaptor<Map<String, String>> headersCaptor;
+    @Captor ArgumentCaptor<UrlConnector> urlConnectorCaptor;
+    @Mock SimpleAmlResult simpleAmlResultMock;
+
+    @Before
+    public void setUp() throws Exception {
+        keyPair = generateKeyPairFrom(KEY_PAIR_PEM);
+
+        when(pathFactoryMock.createAmlPath(SOME_APP_ID)).thenReturn(GENERATED_PATH);
+    }
+
+    @Test
+    public void shouldPerformAmlCheck() throws Exception {
+        when(objectMapperMock.writeValueAsString(amlProfileMock)).thenReturn(SERIALIZED_BODY);
+        when(signatureFactoryMock.create(keyPair.getPrivate(), HTTP_POST, GENERATED_PATH, SERIALIZED_BODY.getBytes())).thenReturn(SOME_SIGNATURE);
+        when(resourceFetcherMock.postResource(any(UrlConnector.class), any(byte[].class), any(Map.class), eq(SimpleAmlResult.class)))
+                .thenReturn(simpleAmlResultMock);
+
+        SimpleAmlResult result = testObj.performCheck(keyPair, SOME_APP_ID, amlProfileMock);
+
+        verify(resourceFetcherMock).postResource(urlConnectorCaptor.capture(), eq(SERIALIZED_BODY.getBytes()), headersCaptor.capture(), eq(SimpleAmlResult.class));
+        assertUrl(urlConnectorCaptor.getValue());
+        assertHeaders(headersCaptor.getValue());
+        assertSame(result, simpleAmlResultMock);
+    }
+
+    private void assertUrl(UrlConnector urlConnector) throws MalformedURLException {
```

Comment:
I not sure that is worth to have all this private method. I think it's better to put the code inside the `shouldPerformAmlCheck()`. And if the problem is the clarity of the checks then it's better to have test variables with more expressive names, like SOME_SIGNATURE ~ TEST_DIGEST and so on. I know that is the same approach or similar of the other test but seems this is a new one I think that we can think about that at least in this class.

Reply:
> I think it's better to put the code inside the shouldPerformAmlCheck()

I tried it inline, it was awful.

> And if the problem is the clarity of the checks then it's better to have test variables with more expressive names

I don't think the variable names are the problem.  It's the boiler plate around reaching what we want to test that is causing a mess.  Maybe hamcrest would clean it up.

Reply:
hamcrest was too much.  See if its better now.